src/pudl/analysis/state_demand.py

```diff
@@ -499,45 +548,61 @@ def load_eia861_state_total_sales(
     return df[["state_id_fips", "year", "demand_mwh"]]
 
 
-def predict_state_hourly_demand(
-    demand: pd.DataFrame,
-    counties: pd.DataFrame,
-    assignments: pd.DataFrame,
-    state_totals: pd.DataFrame = None,
-    mean_overlaps: bool = False,
+@asset(
+    io_manager_key="pudl_sqlite_io_manager",
+    compute_kind="Python",
+    config_schema={
+        "mean_overlaps": Field(
+            bool,
+            default_value=False,
+            description=(
+                "Whether to mean the demands predicted for a county in cases when a "
+                "county is assigned to multiple respondents. By default, demands are "
+                "summed."
+            ),
+        ),
+    },
+)
+def predicted_state_hourly_demand(
+    context,
+    imputed_hourly_demand_ferc714: pd.DataFrame,
+    census_counties: pd.DataFrame,
+    county_assignments_ferc714: pd.DataFrame,
+    total_state_sales_eia861: pd.DataFrame = None,
 ) -> pd.DataFrame:
```

Comment:
In this PR the `clean` and `impute` stages are both long. I grouped the tiny stages (`filter` and `melt`) with their corresponding pair to whittle down to 2 assets, rather than 4. The `load` step currently produces 2 separate assets, rather than 1. I could whittle this down further, but I do think not by much.

The other assets (e.g. `county_assignments`, `census_counties` etc.) perform minor modifications to existing tables and probably could just be called as functions in the state demand asset itself, rather than being defined as intermediate assets. I can double check they don't get called elsewhere and consolidate.

Reply:
Thanks for pruning the intermediate assets! The DAG feels much cleaner now.